pkg/restmapper/caching.go

```diff
@@ -70,12 +72,35 @@ type cachedGVR struct {
 	Scope    meta.RESTScope
 }
 
+// KindFromGVR finds out the Kind from the GVR in the cache. If the GVR version is not given, we will iterate all the matching
```

Comment:
Maybe rename to KindFor, as this is a fairly direct mapping to the KindFor method on RESTMapper

Reply:
I reverted the code unrelated to this PR.